var/spack/repos/builtin/packages/chapel/package.py

```diff
@@ -578,47 +640,51 @@ def setup_env_vars(self, env):
         if self.spec.satisfies("+developer"):
             env.set("CHPL_DEVELOPER", "true")
 
+        if not self.spec.satisfies("llvm=none"):
+            # workaround Spack issue #44746:
+            # Chapel does not directly utilize lua, but many of its
+            # launchers depend on system installs of batch schedulers
+            # (notably Slurm on Cray EX) which depend on a system Lua.
+            # LLVM includes lua as a dependency, but a barebones lua
+            # install lacks many packages provided by a system Lua,
+            # which are often required by system services like Slurm.
+            # Disable the incomplete Spack lua package directory to
+            # allow the system one to function.
```

Comment:
(minor) that issue suggests that adding `depends_on(lua-luaposix)` would solve this problem, although generally I'm unsure if that's the ideal solution:

If I understand, the issue is that these external scheduler executables check for `LUA_PATH` and use it. Ideally we'd have some way to do that whenever the external scheduler is used (vs. for every package that happens to use both the scheduler and depend on lua).

This seems acceptable as a short term fix though.

Reply:
I agree this is an ugly hack and I wish there was a better solution.

FWIW the `lua-luaposix` package alone was insufficient to solve the immediate problem (it just changed one missing lua package error into a second one). IIUC SLURM's plugin architecture means a system SLURM install can depend on a site-specific number of additional lua packages the sysadmins may have installed. Tracking all those potential dependencies in a Chapel package seems inappropriate, since Chapel itself does not directly depend on lua at all. Chapel simply wants to invoke the system SLURM commands and not have them drop dead due to missing packages in Spack's underprovisioned lua install.

Perhaps what we actually want is something like `depends_on("scheduler")`, where `scheduler` is a virtual dependency fulfilled by any number of (external system-provided) scheduler packages that also correctly encode all their transitive dependencies with variants (e.g. a list of specific lua packages). But unless those can be automatically detected with very smart external detection, that likely places the burden on the user running Spack to correctly express all those dependencies.